opcua/common/xmlparser.py

```diff
@@ -206,65 +201,76 @@ def _parse_attr(self, el, obj):
         else:
             self.logger.info("Not implemented tag: %s", el)
 
+    def _parse_contained_value(self, el, obj):
+        """
+        Parse the child of el as a constant.
+        """
+        val_el = el.find(".//")  # should be only one child
+        self._parse_value(val_el, obj)
+
     def _parse_value(self, val_el, obj):
-        child_el = val_el.find(".//")  # should be only one child
-        if child_el is not None:
-            ntag = self._retag.match(child_el.tag).groups()[1]
+        """
+        Parse the node val_el as a constant.
+        """
+        if val_el is not None:
+            ntag = self._retag.match(val_el.tag).groups()[1]
         else:
             ntag = "Null"
-        obj.valuetype = ntag
 
-        if ntag in ("Int8", "UInt8", "Int16", "UInt16", "Int32", "UInt32", "Int64", "UInt64"):
-            obj.value = int(child_el.text)
-        elif ntag in ("Float", "Double"):
-            obj.value = float(child_el.text)
-        elif ntag == "Boolean":
-            obj.value = _to_bool(child_el.text)
+        obj.valuetype = ntag
+        if ntag == "Null":
+            obj.value = None
+        elif ntag in (
+                "SByte", "Byte",
+                "Int16", "UInt16", "Int32", "UInt32", "Int64", "UInt64",
+                "Float", "Double",
+                "Boolean",
+                "DateTime" #FIXME Check format in a specific elif?
```

Comment:
Not sure since ByteString and String are part of ua.ua_binary.Primitives and they are in another elif.
However Primitives1 might do the trick if we treat DateTime in its own elif (might also be prettier for future checks if any).
I'll give it a try